codex-ui/src/vue/components/tabbar/Tabbar.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div class="tabbar">
+    <Tab
+      v-for="tab in tabs"
+      :key="tab.identifier"
+      :is-active="tab.isActive"
+      :title="tab.title"
+      :closable="tab.closable"
+      @close="$emit('tabDiscard', tab.identifier)"
+      @click="$emit('tabClicked', tab.identifier)"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { TabList, TabParams } from './Tab.types';
+import Tab from './Tab.vue';
+
+defineEmits<{
+  /* eslint-disable-next-line no-unused-vars */
+  (e: 'tabDiscard', identifier: TabParams['identifier']): void;
+  /* eslint-disable-next-line no-unused-vars */
+  (e: 'tabClicked', identifier: TabParams['identifier']): void;
+}>();
```

Comment:
lets use the new vue 3.3. [syntax](https://vuejs.org/guide/typescript/composition-api.html#typing-component-emits)

```
const emit = defineEmits<{
  click: [tab: TabParams]
  discard: [tab: TabParams]
}>()
```